src/utils/neu_jwt.c

```diff
@@ -27,45 +29,119 @@
 #include "log.h"
 #include "neu_jwt.h"
 
-char   public_key[4096]  = { 0 };
-char   private_key[4096] = { 0 };
-char   token_local[4096] = { 0 };
+char token_local[4096] = { 0 };
+
+int        n_pub_file = 0;
+key_file_t pub_key_files[256];
+
 size_t priv_key_len      = 0;
-size_t pub_key_len       = 0;
+char   private_key[4096] = { 0 };
 
-int neu_jwt_init(char *pub_key_path, char *priv_key_path)
+static int handle_public_key(char *dir_path, char *public_name)
 {
-    FILE *fp_priv_key = NULL;
-    FILE *fp_pub_key  = NULL;
+    FILE *fp_pub_key           = NULL;
+    char  pub_key_string[4096] = { 0 };
+    char  pub_path[256]        = { 0 };
+    n_pub_file                 = 0;
 
-    fp_priv_key = fopen(priv_key_path, "r");
-    if (fp_priv_key == NULL) {
-        log_error("Failed to open key file: %s, error: %d", priv_key_path,
+    snprintf((char *) pub_path, sizeof(pub_path), "%s/%s", dir_path,
+             public_name);
+
+    fp_pub_key = fopen(pub_path, "r");
+
+    if (fp_pub_key == NULL) {
+        log_error("Failed to open public key file: %s, errno: %d", public_name,
+                  errno);
+        return -1;
+    }
+
+    pub_key_files[n_pub_file].len =
+        fread(pub_key_string, 1, sizeof(pub_key_string), fp_pub_key);
+    if (pub_key_files[n_pub_file].len <= 0) {
+        log_error("Failed to read public key file: %s, errno: %d", public_name,
                   errno);
         return -1;
     }
+
+    strncpy(pub_key_files[n_pub_file].value, pub_key_string,
+            sizeof(pub_key_files[n_pub_file].value));
+
+    memcpy(pub_key_files[n_pub_file].name, public_name,
+           strlen(public_name) - 4);
+
```

Comment:
strncpy

Reply:
why -4

Reply:
public_name is the name of the public key file, I need the name before the ".pem"